include/sparrow/fixed_size_layout.hpp

```diff
@@ -0,0 +1,370 @@
+
+        bitmap_reference has_value(size_type i);
+        bitmap_const_reference has_value(size_type i) const;
+        inner_reference value(size_type i);
+        inner_const_reference value(size_type i) const;
+
+        value_iterator value_begin();
+        value_iterator value_end();
+
+        const_value_iterator value_cbegin() const;
+        const_value_iterator value_cend() const;
+
+        bitmap_iterator bitmap_begin();
+        bitmap_iterator bitmap_end();
+
+        const_bitmap_iterator bitmap_cbegin() const;
+        const_bitmap_iterator bitmap_cend() const;
+
+        friend class reference_proxy<fixed_size_layout>;
+        friend class const_reference_proxy<fixed_size_layout>;
+    };
+
+    /***************************************************
+     * fixed_size_layout_value_iterator implementation *
+     **************************************************/
+
+    template <class T, bool is_const>
+    fixed_size_layout_value_iterator<T, is_const>::fixed_size_layout_value_iterator(pointer pointer)
+        : m_pointer(pointer)
+    {
+    }
+
+    template <class T, bool is_const>
+    auto fixed_size_layout_value_iterator<T, is_const>::dereference() const -> reference
+    {
+        return *m_pointer;
+    }
+
+    template <class T, bool is_const>
+    void fixed_size_layout_value_iterator<T, is_const>::increment()
+    {
+        ++m_pointer;
+    }
+
+    template <class T, bool is_const>
+    void fixed_size_layout_value_iterator<T, is_const>::decrement()
+    {
+        --m_pointer;
+    }
+
+    template <class T, bool is_const>
+    void fixed_size_layout_value_iterator<T, is_const>::advance(difference_type n)
+    {
+        m_pointer += n;
+    }
+
+    template <class T, bool is_const>
+    auto fixed_size_layout_value_iterator<T, is_const>::distance_to(const self_type& rhs) const -> difference_type
+    {
+        return rhs.m_pointer - m_pointer;
+    }
+
+    template <class T, bool is_const>
+    bool fixed_size_layout_value_iterator<T, is_const>::equal(const self_type& rhs) const
+    {
+        return rhs.m_pointer == m_pointer;
+    }
+
+    template <class T, bool is_const>
+    bool fixed_size_layout_value_iterator<T, is_const>::less_than(const self_type& rhs) const
+    {
+        return m_pointer < rhs.m_pointer;
+    }
+
+    /************************************
+     * fixed_size_layout implementation *
+     ***********************************/
+
+    template <class T>
+    fixed_size_layout<T>::fixed_size_layout(array_data ad)
+        : m_data(std::move(ad))
+    {
+        // We only require the presence of the bitmap and the first buffer.
+        assert(m_data.buffers.size() > 0);
+        assert(m_data.length == m_data.buffers[0].size());
+        assert(m_data.length == m_data.bitmap.size());
+    }
+
+    template <class T>
+    auto fixed_size_layout<T>::size() const -> size_type
+    {
+        assert(m_data.buffers.size() > 0);
+        return m_data.buffers[0].size();
```

Comment:
I think you should read `m_data.length` instead. The reason is that `m_data.buffers[0]` may hold `uint8_t` while the data type of the array is `int32_t` for instance. Therefore, `size` would not be what you expect.